pulsar-client/src/main/java/org/apache/pulsar/client/impl/MultiTopicsConsumerImpl.java

```diff
@@ -269,7 +269,7 @@ private void receiveMessageFromConsumer(ConsumerImpl<T> consumer) {
                 // thenAcceptAsync, there is no chance for recursion that would lead to stack overflow.
                 receiveMessageFromConsumer(consumer);
             }
-        }, internalPinnedExecutor).exceptionally(ex -> {
+        }, externalPinnedExecutor).exceptionally(ex -> {
```

Comment:
why change to the external pinned executor, the `externalPinnedExecutor` is used to trigger message listeners.